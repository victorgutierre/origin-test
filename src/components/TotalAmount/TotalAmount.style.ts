import styled from 'styled-components';

export const TotalAmountWrapper = styled.div`
  display: block;
  width: 100%;
  border-radius: 8px;
  border: 1px solid #eaeef2;
`;

export const TotalAmountTopInfo = styled.div`
  display: flex;
  justify-content: space-between;
  padding: 2rem;
`;

export const TotalAmountDescription = styled.p`
  font-size: 1.5rem;
  color: #202a31;
`;

export const TotalAmountValue = styled.p`
  text-align: right;
  font-size: 2rem;
  font-weight: bold;
  color: #2f7af6;
`;

export const TotalAmountDetails = styled.div`
  display: block;
  padding: 2rem;
  background-color: #eaeef2;
`;
