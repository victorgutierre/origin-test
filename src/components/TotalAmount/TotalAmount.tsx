import React from 'react';
import { months, money } from '../../utils/utils';
import {
  TotalAmountWrapper,
  TotalAmountTopInfo,
  TotalAmountDetails,
  TotalAmountDescription,
  TotalAmountValue
} from './TotalAmount.style';

interface Props {
  year: number;
  month: number;
  differenceYearsInMonths: number;
  defaultTotalAmount: string;
}

export const TotalAmount: React.FC<Props> = ({
  year,
  month,
  differenceYearsInMonths,
  defaultTotalAmount
}) => {
  const totalValue = money(defaultTotalAmount / differenceYearsInMonths); 

  return (
    <TotalAmountWrapper>
      <TotalAmountTopInfo>
        <TotalAmountDescription>Monthly amount</TotalAmountDescription>
        <TotalAmountValue>${totalValue}</TotalAmountValue>
      </TotalAmountTopInfo>

      <TotalAmountDetails>
        <p>
          You're planning <b>{differenceYearsInMonths} monthly deposits</b> to
          reach your <b>${money(defaultTotalAmount)}</b> goal by{' '}
          <b>
            {months[month]} {year}.
          </b>
        </p>
      </TotalAmountDetails>
    </TotalAmountWrapper>
  );
};
