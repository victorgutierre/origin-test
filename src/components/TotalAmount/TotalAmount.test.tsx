import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';
import { render } from '@testing-library/react';
import { TotalAmount } from './TotalAmount';

describe('<TotalAmount />', () => {
  const actualDate = new Date();
  const year = actualDate.getFullYear();
  const month = actualDate.getMonth();
  const defaultYearPlanning = 4;

  const differenceYearsInMonths = defaultYearPlanning * 12;

  const component = (
    <TotalAmount
      month={month}
      year={year + defaultYearPlanning}
      differenceYearsInMonths={differenceYearsInMonths}
      defaultTotalAmount={25000}
    />
  );

  it('Should render the TotalAmount component', () => {
    const wrapper = shallow(component);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('Date element must render', () => {
    const { queryByTestId } = render(component);
    const date = queryByTestId('total-amount-date');
    expect(date).toBeTruthy();
  });

  it('Date element text should be April 2025.', () => {
    const { queryByTestId } = render(component);
    const date = queryByTestId('total-amount-date');
    expect(date.textContent).toBe('April 2025.');
  });

  it('Total amount monthly amount element must render', () => {
    const { queryByTestId } = render(component);
    const date = queryByTestId('total-amount-monthly-amount');
    expect(date).toBeTruthy();
  });

  it('Total amount value element text should be April 2025', () => {
    const { queryByTestId } = render(component);
    const date = queryByTestId('total-amount-value');
    expect(date).toBeTruthy();
  });

  it('Total amount value element text should be $25,000.00', () => {
    const { queryByTestId } = render(component);
    const date = queryByTestId('total-amount-value');
    expect(date.textContent).toBe('$25,000.00');
  });

  it('Total amount years in months element must render', () => {
    const { queryByTestId } = render(component);
    const date = queryByTestId('total-amount-yearsInMonths');
    expect(date).toBeTruthy();
  });
});
