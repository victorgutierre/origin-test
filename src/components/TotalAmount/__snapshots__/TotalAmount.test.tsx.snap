// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TotalAmount /> Should render the TotalAmount component 1`] = `
<styled.div>
  <styled.div>
    <styled.p>
      Monthly amount
    </styled.p>
    <styled.p
      data-testid="total-amount-monthly-amount"
    >
      $
      521.00
    </styled.p>
  </styled.div>
  <styled.div>
    <p>
      You
      &apos;
      re planning
       
      <b
        data-testid="total-amount-yearsInMonths"
      >
        48
         monthly deposits
      </b>
       
      to reach your
       
      <b
        data-testid="total-amount-value"
      >
        $
        25,000.00
      </b>
       
      goal by
       
      <b
        data-testid="total-amount-date"
      >
        April
         
        2025
        .
      </b>
    </p>
  </styled.div>
</styled.div>
`;
